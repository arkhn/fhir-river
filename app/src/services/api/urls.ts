let API_URL = "" as string | undefined;
if (process.env.NODE_ENV === "production") {
  ({ REACT_APP_API_URL: API_URL } = (window as any).env);
} else {
  ({ REACT_APP_API_URL: API_URL } = process.env);
}

export { API_URL };
