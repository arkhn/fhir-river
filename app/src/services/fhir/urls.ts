export const { REACT_APP_FHIR_API_URL: FHIR_API_URL } = process.env;
