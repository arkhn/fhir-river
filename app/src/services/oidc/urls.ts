export const {
  REACT_APP_OIDC_LOGIN_URL: OIDC_LOGIN_URL,
  REACT_APP_OIDC_LOGOUT_URL: OIDC_LOGOUT_URL,
} = process.env;
