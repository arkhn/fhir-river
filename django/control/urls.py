from django.urls import include, path

from control.api import views
from control.api.router import router

urlpatterns = [
    path("", include(router.urls)),
    path("preview/", views.PreviewEndpoint.as_view(), name="preview"),
    path("delete-resources/", views.ResourceEndpoint.as_view(), name="delete-resources"),
]
