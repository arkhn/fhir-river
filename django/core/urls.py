from django.urls import include, path

from core.api import views
from core.api.router import router

urlpatterns = [
    path("", include(router.urls)),
    path("version/", views.VersionEndpoint.as_view()),
]
