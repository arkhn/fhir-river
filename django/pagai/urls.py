from django.urls import path

from pagai.api import views

urlpatterns = [
    path("pagai/list-owners/", views.OwnersListView.as_view(), name="list-owners"),
    path("pagai/owner-schema/<owner>/", views.OwnerSchemaView.as_view(), name="owner-schema"),
    path("pagai/explore/<resource_id>/<owner>/<table>/", views.ExploreView.as_view(), name="explore"),
]
