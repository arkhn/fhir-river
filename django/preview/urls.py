from django.urls import include, path

from preview.api.router import router
from preview.api import views

urlpatterns = [
    path("", include(router.urls)),
    path("preview/", views.PreviewEndpoint.as_view(), name="preview"),
]
