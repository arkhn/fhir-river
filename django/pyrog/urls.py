from django.urls import include, path

from pyrog.api import views
from pyrog.api.router import router


urlpatterns = [
    path("api/", include(router.urls)),
    path('session/', views.Session.as_view()),
    path('oidc/', include('mozilla_django_oidc.urls')),
]
