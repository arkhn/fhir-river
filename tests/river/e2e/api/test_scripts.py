import pytest

from django.urls import reverse

pytestmark = pytest.mark.django_db


def test_list_scripts(api_client):
    url = reverse("scripts")

    response = api_client.get(url)

    assert response.status_code == 200
    assert response.data == [
        {"category": "cleaning", "description": "Map (0,1) to (False, True)", "name": "binary_to_bool_1"},
        {"category": "cleaning", "description": "Map (0,1) to (True, False)", "name": "binary_to_bool_2"},
        {"category": "cleaning", "description": None, "name": "clean_address"},
        {
            "category": "cleaning",
            "description": 'Remove terminology system from code ("HL7:male") to ("male")',
            "name": "clean_codes",
        },
        {"category": "cleaning", "description": None, "name": "clean_date"},
        {"category": "cleaning", "description": None, "name": "clean_dateTime"},
        {"category": "cleaning", "description": None, "name": "clean_instant"},
        {"category": "cleaning", "description": None, "name": "clean_phone"},
        {
            "category": "cleaning",
            "description": "[deprecated: river parses types automatically]\n"
            "Removes input not conform to FHIR quantity type",
            "name": "clean_quantity",
        },
        {"category": "cleaning", "description": None, "name": "clean_time"},
        {
            "category": "cleaning",
            "description": "Return None when entry is some code: -1 or (sans)",
            "name": "code_to_empty",
        },
        {"category": "cleaning", "description": None, "name": "if_valid"},
        {"category": "cleaning", "description": "Capitalize and strip", "name": "make_title"},
        {
            "category": "cleaning",
            "description": "Map some code from (2->True) and None otherwise",
            "name": "map_2_true",
        },
        {"category": "cleaning", "description": "Map (O,N) to (True, False)", "name": "map_deceased"},
        {"category": "cleaning", "description": "Maps French family situation", "name": "map_family_situation"},
        {
            "category": "cleaning",
            "description": "Map gender from (M,F) or (HL7:M, HL7:F) to (male,female)",
            "name": "map_gender",
        },
        {
            "category": "cleaning",
            "description": "Map gender from (1,2) to (male, female)",
            "name": "map_gender_numeric",
        },
        {"category": "cleaning", "description": "Map (0,1,NULL) to (plan, order, proposal)", "name": "map_intent"},
        {"category": "cleaning", "description": "Map MIMIC marital status", "name": "map_marital_status"},
        {
            "category": "cleaning",
            "description": "Map UMLS codes (Yes, No) to (permit, deny)",
            "name": "map_permission",
        },
        {
            "category": "cleaning",
            "description": "Map int to ServiceRequest.priority (0: stat, 1: asap, 2: " "urgent, 3+ routine)",
            "name": "map_priority",
        },
        {"category": "cleaning", "description": "Map code (0,1) to (active,inactive)", "name": "map_status"},
        {
            "category": "cleaning",
            "description": 'Convert ("True","true","TRUE") to (True)..',
            "name": "string_to_bool",
        },
        {
            "category": "cleaning",
            "description": "Strip strings, convert NaN and None to empty string",
            "name": "strip",
        },
        {"category": "cleaning", "description": "Return None when entry is 0", "name": "zero_to_empty"},
        {
            "category": "merging",
            "description": 'Merging script with a simple concatenation and a " " ' "separator",
            "name": "merge_concat",
        },
        {
            "category": "merging",
            "description": "Merging script with a datetime concatenation",
            "name": "merge_datetime",
        },
        {"category": "merging", "description": None, "name": "merge_insee"},
        {
            "category": "merging",
            "description": "Merge two binary entries and return a FHIR CarePlan.status",
            "name": "merge_status",
        },
        {
            "category": "merging",
            "description": "Merging script which select the first input not empty",
            "name": "select_first_not_empty",
        },
    ]
