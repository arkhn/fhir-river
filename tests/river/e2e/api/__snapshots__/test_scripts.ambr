# name: test_list_scripts
  <class 'list'> [
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map (0,1) to (False, True)',
      'name': 'binary_to_bool_1',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map (0,1) to (True, False)',
      'name': 'binary_to_bool_2',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'clean_address',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Remove terminology system from code ("HL7:male") to ("male")',
      'name': 'clean_codes',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'clean_date',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'clean_dateTime',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'clean_instant',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Remove special character from long BLOB / CLOB text in ORBIS',
      'name': 'clean_long_text_ORBIS',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'clean_phone',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': '
        [deprecated: river parses types automatically]
        Removes input not conform to FHIR quantity type
      ',
      'name': 'clean_quantity',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'clean_time',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Return None when entry is some code: -1 or (sans)',
      'name': 'code_to_empty',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': None,
      'name': 'if_valid',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Capitalize and strip',
      'name': 'make_title',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map some code from (2->True) and None otherwise',
      'name': 'map_2_true',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map (O,N) to (True, False)',
      'name': 'map_deceased',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Maps French family situation',
      'name': 'map_family_situation',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map gender from (M,F) or (HL7:M, HL7:F) to (male,female)',
      'name': 'map_gender',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map gender from (1,2) to (male, female)',
      'name': 'map_gender_numeric',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map (0,1,NULL) to (plan, order, proposal)',
      'name': 'map_intent',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map MIMIC marital status',
      'name': 'map_marital_status',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map UMLS codes (Yes, No) to (permit, deny)',
      'name': 'map_permission',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map int to ServiceRequest.priority (1: stat, 2: asap, 3: urgent, else: routine)',
      'name': 'map_priority',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Map code (0,1) to (active,inactive)',
      'name': 'map_status',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Convert ("True","true","TRUE") to (True)..',
      'name': 'string_to_bool',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Strip strings, convert NaN and None to empty string',
      'name': 'strip',
    },
    <class 'dict'> {
      'category': 'cleaning',
      'description': 'Return None when entry is 0',
      'name': 'zero_to_empty',
    },
    <class 'dict'> {
      'category': 'merging',
      'description': 'Merging script with a simple concatenation and a " " separator',
      'name': 'merge_concat',
    },
    <class 'dict'> {
      'category': 'merging',
      'description': 'Merging script with a datetime concatenation',
      'name': 'merge_datetime',
    },
    <class 'dict'> {
      'category': 'merging',
      'description': None,
      'name': 'merge_insee',
    },
    <class 'dict'> {
      'category': 'merging',
      'description': 'Merging script with a instant concatenation',
      'name': 'merge_instant',
    },
    <class 'dict'> {
      'category': 'merging',
      'description': 'Merge two binary entries and return a FHIR CarePlan.status',
      'name': 'merge_status',
    },
    <class 'dict'> {
      'category': 'merging',
      'description': 'Merging script which select the first input not empty',
      'name': 'select_first_not_empty',
    },
  ]
---
